edpop_explorer/edpopxshell.py

```diff
@@ -120,7 +137,7 @@ def _show_records(self, records: List[APIRecord],
             return 0
         # Determine count (the number of items to show)
         if limit is None:
-            count = limit
+            count = remaining
         else:
             count = min(remaining, limit)
```

Comment:
If you set the default value of `limit` as `math.inf` instead of `None`, you can just use `count = min(remaining, limit)` in all cases. That's a matter of preference though.

Reply:
Done! Unfortunately this doesn't work well with typing hints though, so I removed that here.